ISISScientificComputing/autoreduce
# Extract requirements from setup.py into requirements.txt

### Summary of work

Extracts requirements from setup.py into `requirements.txt`.

Move the dev requirements into `dev-requirements.txt`

This is done because for whatever reason `pip install -e autoreduce` will upgrade packages that autoreduce **DOES NOT** depend on.

`pip install -r` does not have this behaviour.

This was causing issues on production by thrashing the `h5py` needed by Mantid.

### How to test your work
- Code review the changes to the actions


The fastest way to see the result is to pull the docker container (adjust autoreduce code location).
```
docker run --network=host -v ~/dev/autoreduce/:/autoreduce -v /isis:/isis -it autoreduction/qp /bin/bash
```

To test the installation:
- Create a new virtual or conda env
- `pip install -r requirements.txt && pip install -r dev-requirements.txt`
- AR should run as normal
